rke/templates/templates.go

```diff
@@ -167,7 +168,8 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			">=1.8.0-rancher0 <1.16.0-alpha":         weavev18,
 		},
 		kdm.Aci: {
-			">=1.17.0-alpha": aciv500,
+			">=1.17.0-alpha <1.20.0-rancher0": aciv500,
+			">=1.20.0-rancher0 <1.23.0-rancher1-1": aciv523,
```

Comment:
A lot more conditions need to be introduced here to ensure that updated `aciv523` template is used for newly introduced versions only. With the current condition (`>=1.20.0-rancher0 <1.23.0-rancher1-1`) it will be used for any k8s version starting with 1.20, but previously existing versions (e.g. `v1.20.15-rancher1-4`) reference 5.1.1.0 images and thus cannot use the new template that is built for 5.2.3.2.
Please see below an example for `kdm.NginxIngress` that demonstrates more granular conditions to ensure only newly added versions use new template.

Reply:
Updated required conditions in [7e7eabb](https://github.com/rancher/kontainer-driver-metadata/pull/922/commits/7e7eabbba1b4688c0359d90daf1072e09a1fe64f)